bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Caret.java

```diff
@@ -132,7 +132,9 @@ Rectangle getBoundsInPixels () {
 	if (width == 0) {
 		int [] buffer = new int [1];
 		if (OS.SystemParametersInfo (OS.SPI_GETCARETWIDTH, 0, buffer, 0)) {
-			return new Rectangle (getXInPixels(), getYInPixels(), buffer [0], getHeightInPixels());
+			int width = DPIUtil.pixelToPoint(buffer [0], Win32DPIUtils.getPrimaryMonitorZoomAtStartup());
+			int widthInPixels = Win32DPIUtils.pointToPixel(width, getNativeZoom());
```

Comment:
Maybe these two lines (or even the whole logic for retrieving the size with `SystemParameterInfo` could be extracted into a common utility method.